firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/vector.txt

```diff
@@ -0,0 +1,60 @@
+Scenario: The function in Util Vectors works correctly
+  GIVEN("The util resize function")
+  WHEN("the util resize function is called on 252UL buffer size")
+    THEN("Result is as expected ")
+
+  WHEN("Util function empty is used on vector")
+    THEN("Result is as expected")  
+
+  WHEN("Util function avaliable is used on vector")
+    THEN("Result is as expected")  
+
+  GIVEN("Util full function")   
+  WHEN("Util function full is used on vector")
+    THEN("Result is as expected")
+
+    
+Scenario: The function in Util push_back works correctly
```

Comment:
The `push_back` method should be tested with the following possible conditions:
- The `Vector` has more than one space available before `push_back`. `push_back` should succeed, and the next `push_back` should also succeed.
- The `Vector` has one space available before `push_back`. `push_back` should succeed, and the next `push_back` should fail with a bounds error.
- The `Vector` is completely full before `push_back`. `push_back` should fail with a bounds error. Afterwards, the `Vector` should behave as if the `push_back` method hadn't been called at all.

In every condition, after the `push_back`:

- The Vector should correctly report its size (is it empty? is it full? how much capacity is available? what is its size?).
- All indices which we expect to be out-of-bounds (e.g. for erase) should actually be out-of-bounds, while all indices which we expect to be in-bounds (e.g. for erase) should actually be in-bounds.
- All elements which were previously in the `Vector` should remain unchanged.
- If `push_back` succeeded, a new element should have been added to the end of the vector, and that element should have the value given as input to `push_back`, while every other element in the vector remains unchanged.